scipy/optimize/_basinhopping.py

```diff
@@ -630,10 +630,17 @@ def basinhopping(func, x0, niter=100, T=1.0, stepsize=0.5,
             break
 
     # prepare return object
-    lowest = bh.storage.get_lowest()
+    lowest_minres = bh.storage.get_lowest()
     res = bh.res
-    res.x = np.copy(lowest[0])
-    res.fun = lowest[1]
+    res.x = np.copy(lowest_minres.x)
+    res.fun = lowest_minres.fun
+
+    if hasattr(lowest_minres, "jac"):
+        res.jac = np.copy(lowest_minres.jac)
+
+    if hasattr(lowest_minres, "hess"):
+        res.hess = np.copy(lowest_minres.hess)
+
```

Comment:
Rather than the above few lines we could think about doing

    res.lowest_optimization_result = lowest_minres

Then people can access `jac` and `hess` via the optimization result.